lib/index.js

```diff
@@ -40,18 +23,53 @@ S3rver.prototype.resetFs = function (callback) {
   });
 }
 
+S3rver.prototype.callback = function () {
+  return new App(this.options);
+}
+
 S3rver.prototype.run = function (done) {
   const app = new App(this.options);
-  const server = app.serve(done);
-  if (this.options.removeBucketsOnClose) {
-    const close = server.close.bind(server);
-    server.close = (...args) => {
-      this.resetFs(() => {
-        close(...args);
-      });
-    };
-  }
+  let server = ((this.options.key && this.options.cert) || this.options.pfx)
+      ? https.createServer(this.options, app)
+      : app;
+  server = server.listen(this.options.port, this.options.hostname, (err) => {
+    done(err, this.options.hostname, this.options.port, this.options.directory);
+  }).on('error', (err) => {
+    done(err);
+  });
+  server.close = (callback) => {
+    const { close } = Object.getPrototypeOf(server);
+    return close.call(server, () => {
+      app.logger.unhandleExceptions();
+      app.logger.close();
+      if (this.options.removeBucketsOnClose) {
+        this.resetFs(callback);
+      } else {
+        callback();
+      }
+    });
+  };
   return server;
 };
 
+S3rver.defaultOptions = {
+  port: 4578,
+  hostname: 'localhost',
+  silent: false,
+  cors: {
+    CORSConfiguration: {
+      CORSRule: [{
+        AllowedOrigin: ['*'],
+        AllowedMethod: ['GET'],
```

Comment:
I believe that's correct? If you think that's problematic/confusing I'm ok with disabling CORS by default. I think most people needing to enable CORS are going to have their own config anyway.

In case you're wondering, this is just copy and pasted from the result of parsing the XML config you get after creating a new bucket.

Reply:
Just wanted to be certain about what these defaults do. I don't think most users will be expecting CORS protection to be enabled by default (this could be a point of frustration for them) but at the same time since most users are using this for integration testing it should resemble the real-world S3 as closely as possible. I'm partial towards the latter so I support leaving it as is.